### ISSUE TITLE
MatchError when attempting to use parameterized type from another module

### ISSUE THREAD

**Issue by stocks29** [pylon/spect#12]
When attempting to use a parameterized type from a remote module a `MatchError` is throw. Parameterized types in the local module seem to work fine. Example below:

```elixir
defmodule Types do
  @type maybe(t) :: t | nil
end
```

```elixir
defmodule OtherModule do
  @type bar :: Types.maybe(binary()) 
end
```

Error:
```elixir
 {:error, %MatchError{term: [{:atom, 0, Types}, {:atom, 0, :maybe}, [{:type, 40, :binary, []}]]}}
```

Really loving `Spect` by the way! Thanks for creating it!

**Comment by brentspell** [comment]
Hi stocks. You are correct there, spect doesn't currently support parameterized types. In fact, we weren't even aware that such things existed! I'll take a look and see what I can do to add support.

**Comment by brentspell** [comment]
Thanks for the report. This should be fixed in release 0.4.0.

**Comment by stocks29** [comment]
thanks for the fast turnaround! this is great :)
